docs/DeveloperGuide.md

```diff
@@ -18,12 +44,162 @@
 
 {Describe the value proposition: what problem does it solve?}
 
+
+## Implementations
+### Exit Command
+
+The `ExitCommand` inherits its properties from the abstract `Command` class.
+
+The command to exit the program is `exit`.
+If 'exit' is called, the program prints the exit message and terminates the program.
+
+This is how the ExitCommand works:
+
+1. The `main()` method in Main calls `run()` in Main. 
+
+2. `run()` will call `runCommandTillExit()`.
+
+3. The `ui` reads the user's input to retrieve
+the command via `ui.getUserInput()` and parses it through `commandParser.parseCommand()`.
+
+4. Within the `commandParser.parseCommand()`, some functions are also internally called.
+    1. `newCommand()` is called to identify the String received
+    2. `parseKeyword()` is called to split the command from the parameters.
+ 
+5. A new `ExitCommand()` is called.
+
+6. `ExitCommand` is returned to parseCommand().
+
+7. `ExitCommand` is returned to main().
+
+8. The Exit Command is now executed via `command.execute()`.
+
+9. `runCommandTillExit()` will now call `command.isExit()` which returns `true`.
+
+10. The loop is now broken and ui will call `ui.showEndingMessage()`.
+
+11. The program will now exit.
+
+<p align="center">
+    <img src="images/ExitCommandSequenceDiagram.png">
+    <br />
+    <i>Figure 2: Sequence Diagram for Bye Command</i>
+</p>
+
+### List Pet Command
+
+The `ListPetCommand` inherits its properties from the abstract `Command` class.
+
+The command to list all pets currently available is `list`. When called, the program prints all the names and statistics of the existing pets (if any), followed by the number of pets currently in the list.
+
+This is how the ListPetCommand works:
+
+1. The `main()` method calls `ui.getUserInput()` from `ui`. The user's command is parsed through `commandParser.parseCommand()`.
+
+2. Within `commandParser.parseCommand()`, `newCommand()` will identify the command received (`list`).
+
+3. `ListPetCommand()` calls the `list()` method from the `PetList` class.
+
+4. `PetList.list()` then iterates through all pets in `petList`, calling `pet.getPetName()`, `pet.getPetType()`, `pet.getAge()` and `pet.getWeight()` each iteration, then prints the number of pets.
+
+5. The loop is now broken to return `ListPetCommand` to `parseCommand()`.
+
+6. `ListPetCommand` is returned to `main()`.
+
+7. The program is now ready to receive another command.
+
+<p align="center">
+    <img src="images/ListPetCommand.png">
+    <br />
+    <i>Figure 3: Sequence Diagram for List Pet Command</i>
+</p>
```

Comment:
Perhaps the text is missing from :commandParser to :Main.
In addition, it seems that the return from :PetList to command:ListPetCommand is missing.